### PR TITLE
Adapting downloaded Qiskit sample

### PR DESCRIPTION
**Pull request by ffrachon**
If you have some experience with quantum computing or are just starting, it is likely that you will use some samples from Qiskit.org. This Jupyter notebook walks you through four simple steps to adapt an existing Qiskit sample to run against any of the Azure Quantum back-ends.

### CODE REVIEW

File: samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+          "transient": {
+            "deleting": false
+          }
+        }
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# 3. Replace the aer_simulator with an Azure Quantum backend\r\n",
```
Comment (xinyi-joffre) [inline samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb:61]: Should we clarify that by replacing a local simulator with a cloud simulator (like ionq or rigetti) will have differences in performance and quota? For instance, there's lots of samples where you can run many passes on AER locally within seconds, but you won't be able to just replace the backend and get similar behavior.

When working with students at last hackathon it was not clear to them that AER is a local only simulator and does not submit cloud jobs, and it was not clear to them where to find cloud job info in Job Management blade.

File: samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        "\r\n"
+      ],
+      "metadata": {
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      }
+    },
```
Comment (xinyi-joffre) [inline samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb:96]: Should we add instructions on how to install additional packages if needed using !pip install syntax? Some of the qiskit libs for ML use pytorch. That may need to be installed based on what they are trying to do. Pytorch itself can be 1GB by itself, and more if you pull in optional packages, so it's not great to add to the base image, unless we have a lot of usage of pytorch.

Comment (ffrachon) [reply samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb:106]: Thanks for the suggestions. Made the changes.

File: samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        }
+      },
+      "source": [
+        "# Adapting Qiskit samples to run on Azure Quantum\n",
+        "\n",
+        "If you have some experience with quantum computing or are just starting, it is likely that you will use some samples from Qiskit.org. This Jupyter notebook walks you through 4 simple steps to adapt an existing Qiskit sample to run against any of the Azure Quantum back-ends.\n",
+        "\n",
+        "Most samples are configured to run by default against the aer_simulator, which is a great way to get started.\n",
+        "\n",
+        "The aer_simulator is a local simulator, extremely performant, but different from a cloud simulator. When a local simulator is used, all jobs are computed \"in memory\" and there is no network latency nor job queue compared to an actual cloud back-end (simulator or QPU). It enables you to run these simulations very quickly but they may not represent accurately an actualy back-end behavior.\n",
```
Comment (xinyi-joffre) [inline samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb:20]:
```suggestion
        "The aer_simulator is a local simulator, extremely performant, but different from a cloud simulator. When a local simulator is used, all jobs are computed \"in memory\" and there is no network latency nor job queue compared to an actual cloud back-end (simulator or QPU). It enables you to run these simulations very quickly, but they may not accurately represent the behavior of a cloud back-end.\n",
```

File: samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        "\n",
+        "When you don't use a local simulator, you may experience total experiment duration that are orders of magnitude longer (from minutes for a local simulator, to hours or days for an actual back-end). \n",
+        "\n",
+        "Additionally a local simulator only uses the 'local' compute or a hosted compute. When using a cloud back-end you may be spending money or credits, depending what you selected.\n",
```
Comment (xinyi-joffre) [inline samples/azure-quantum/adapting-code/adapting-qiskit-sample.ipynb:24]:
```suggestion
        "Additionally a local simulator only uses the 'local' compute or a hosted compute. When using a cloud back-end you may be spending money or credits, depending on what you selected.\n",
```

**Review by xinyi-joffre — APPROVED** [review]

**Pull request merged**